### ISSUE TITLE
Settle on camelCase over kebab-case in template contexts

### ISSUE THREAD

**Issue by vvagaytsev** [garden-io/garden#3513]

**Comment by hnicke** [comment]
According to the [migration guide](https://docs.garden.io/tutorials/migrating-to-bonsai) this has landed with 0.13.
However it looks like this renaming has not been done yet, right?
The migration guide should probably get adjusted.

**Comment by vvagaytsev** [comment]
Good catch, @hnicke! Thanks for reporting this! Yes, this change hasn't been addressed in 0.13. We'll revisit and update the migration guide.

**Comment by vvagaytsev** [comment]
A follow-up PR is #4825.
